utilities/network.py

```diff
@@ -1038,23 +1038,133 @@ def get_cluster_cni_type(admin_client):
+
+    if not _node_matches_policy_selector(node=node, policy=policy):
+        LOGGER.info(
+            f"Skipping verification for node {sriov_node_network_state.name} - "
+            f"does not match policy {policy.name} nodeSelector"
+        )
+        return
+
+    policy_spec = policy.instance.spec
+    pf_names = policy_spec.nicSelector.get("pfNames", [])
+    root_devices = policy_spec.nicSelector.get("rootDevices", [])
+    expected_num_vfs = policy_spec.get("numVfs")
+    expected_mtu = policy_spec.get("mtu")
+
+    matching_interfaces = []
+    for iface in sriov_node_network_state.instance.status.interfaces:
+        if iface.name in pf_names or iface.pciAddress in root_devices:
+            matching_interfaces.append(iface)
+
+    if not matching_interfaces:
+        LOGGER.info(
+            f"Skipping verification for node {sriov_node_network_state.name} - "
+            f"no interfaces matching policy {policy.name} (pfNames={pf_names}, rootDevices={root_devices})"
+        )
+        return
+
+    for iface in matching_interfaces:
+        LOGGER.info(
+            f"Verifying interface {iface.name} on node {sriov_node_network_state.name}: "
+            f"numVfs={iface.numVfs} (expected: {expected_num_vfs})"
+        )
+
+        if iface.numVfs != expected_num_vfs:
+            LOGGER.error(
+                f"Interface {iface.name} on {sriov_node_network_state.name} has numVfs={iface.numVfs}, "
+                f"expected {expected_num_vfs}"
+            )
+            raise AssertionError(f"numVfs mismatch for {iface.name}: got {iface.numVfs}, expected {expected_num_vfs}")
+
+        if expected_mtu:
+            iface_mtu = getattr(iface, "mtu", None)
+            if iface_mtu and iface_mtu != expected_mtu:
+                LOGGER.error(
+                    f"Interface {iface.name} on {sriov_node_network_state.name} has mtu={iface_mtu}, "
+                    f"expected {expected_mtu}"
+                )
+                raise AssertionError(f"MTU mismatch for {iface.name}: got {iface_mtu}, expected {expected_mtu}")
+
+        LOGGER.info(f"Interface {iface.name} configuration verified successfully")
+
+
+def wait_for_ready_sriov_nodes(snns, policy=None):
+    """
+    Wait for SR-IOV nodes to be ready and verify interface configuration matches policy.
+
+    Args:
+        snns: List of SriovNetworkNodeState objects
+        policy: Optional SriovNetworkNodePolicy object to verify configuration against
+
+    Raises:
+        TimeoutExpiredError: If sync status doesn't reach SUCCEEDED state or configuration doesn't match
+        AssertionError: If interface configuration doesn't match policy
+    """
+    for sriov_node_network_state in snns:
+        LOGGER.info(f"Waiting for node {sriov_node_network_state.name} to reach SUCCEEDED status")
+        sriov_node_network_state.wait_for_status_sync(wanted_status=SriovNetworkNodePolicy.Status.SUCCEEDED)
```

Comment:
I think it makes more sense now to move this check inside the sampler.
Why? Because we might get here before the SNNS changes states, and the `Succeeded` statue is actually from before this policy started to be applied.
Checking the status, along with verifying the configuration, necessarily means that the `Succeded` status is authentic.